apps/cms/src/lib/cms/config.ts

```diff
@@ -0,0 +1,200 @@
+
+export type StrapiConfig = CreateProviderConfig<"strapi">;
+export type ContentfulConfig = CreateProviderConfig<"contentful">;
+export type DatocmsConfig = CreateProviderConfig<"datocms">;
+
+export const channelsConfigSchema = z.object({
+  enabledInChannels: z.array(z.string()),
+});
+
+export const strapiConfigSchema = z.object({
+  name: z.string().min(1),
+  token: z.string().min(1),
+  baseUrl: z.string().min(1),
+});
+
+export const contentfulConfigSchema = z.object({
+  name: z.string().min(1),
+  token: z.string(),
+  baseUrl: z.string(),
+  environment: z.string(),
+  spaceId: z.string(),
+  locale: z.string(),
+  contentId: z.string(),
+});
+
+export const datocmsConfigSchema = z.object({
+  name: z.string().min(1),
+  token: z.string().min(1),
+  baseUrl: z.string(),
+  environment: z.string(),
+  itemTypeId: z.string().min(1),
+});
+
+export const providerCommonSchema = z.object({
+  id: z.string(),
+  providerName: z.string(),
+});
+
+export type ProviderCommonSchema = z.infer<typeof providerCommonSchema>;
+
+export const channelCommonSchema = z.object({
+  channelSlug: z.string(),
+});
+
+export type ChannelCommonSchema = z.infer<typeof channelCommonSchema>;
+
+// todo: helper function so you dont have to merge manually
+export const providersSchemaSet = {
+  strapi: strapiConfigSchema.merge(providerCommonSchema),
+  contentful: contentfulConfigSchema.merge(providerCommonSchema),
+  datocms: datocmsConfigSchema.merge(providerCommonSchema),
+};
+
+export type CMSProviderSchema = keyof typeof providersSchemaSet;
+
+export const providersSchema = z.object(providersSchemaSet);
+
+export type ProvidersSchema = z.infer<typeof providersSchema>;
+
+export type SingleProviderSchema = ProvidersSchema[keyof ProvidersSchema] & ProviderCommonSchema;
+
+export const providerInstanceSchema = z.union([
+  strapiConfigSchema.merge(providerCommonSchema),
+  contentfulConfigSchema.merge(providerCommonSchema),
+  datocmsConfigSchema.merge(providerCommonSchema),
+]);
+
+export type ProviderInstanceSchema = z.infer<typeof providerInstanceSchema>;
+
+export const channelSchema = z
+  .object({
+    enabledProviderInstances: z.array(z.string()),
+  })
+  .merge(channelCommonSchema);
+
+export type ChannelSchema = z.infer<typeof channelSchema>;
+
+export type SingleChannelSchema = ChannelSchema & ChannelCommonSchema;
+
+export type MergedChannelSchema = SingleChannelSchema & {
+  channel: ChannelFragment;
+};
+
+export type CMSChannelSchema = keyof ChannelSchema;
+
+export const cmsSchemaProviderInstances = z.record(z.string(), providerInstanceSchema);
+export const cmsSchemaChannels = z.record(z.string(), channelSchema);
+export const cmsSchema = z.object({
+  providerInstances: cmsSchemaProviderInstances,
+  channels: cmsSchemaChannels,
+});
+
+export type CMSSchemaProviderInstances = z.infer<typeof cmsSchemaProviderInstances>;
+export type CMSSchemaChannels = z.infer<typeof cmsSchemaChannels>;
+export type CMSSchema = z.infer<typeof cmsSchema>;
```

Comment:
Two suggestions here:
1. That's quite a lot of exported types. Please make sure all of this is needed and must be exported.
2. I suggest moving the "channel-related" types and schemas into a separate file. I believe this way, the overhead for a new user who is, f.e. trying to add a new provider, can be a bit lower.